src/daemon/auth/providers/openid_connect/provider.rs

```diff
@@ -587,20 +616,61 @@ impl AuthProvider for OpenIDConnectAuthProvider {
                 // see if we can decode, decrypt and deserialize the users token
                 // into a login session structure
                 let session = self.session_cache.decode(token, &self.session_key)?;
-
                 let status = session.status();
 
-                let mut new_auth = None;
-                if status != SessionStatus::Active {
-                    new_auth = self.try_refresh_token(&session);
-                    if new_auth.is_none() && status == SessionStatus::Expired {
-                        return Err(Error::ApiInvalidCredentials(
-                            "Session expired, please login again".to_string(),
-                        ));
-                    }
+                // Token found in cache and active; all good, do an early return
+                if status == SessionStatus::Active {
+                    return Ok(Some(ActorDef::user(session.id, session.attributes, None)));
                 }
 
-                Ok(Some(ActorDef::user(session.id, session.attributes, new_auth)))
+                let new_auth = match self.try_refresh_token(&session) {
+                    Ok(auth) => {
+                        trace!(
+                            "OpenID Connect: Succesfully refreshed token for user \"{}\"",
+                            &session.id
+                        );
+                        auth
+                    }
+                    Err(err) => {
+                        trace!("OpenID Connect: rfc-6749 5.2 Error Response returned...");
+                        debug!(
+                            "OpenID Connect: Refreshing the token for user '{}' failed: {}",
+                            &session.id, &err
+                        );
+                        match err {
```

Comment:
Minor point: I would prefer not to use initial capitalization for terms such as Session and Authentication Problem within these error strings. Note: these messages will only appear in the Krill log, the actual messages presented to end users are controlled by `en.json` etc in the Lagosta web UI code (assuming that all the used error labels have assigned values in the Lagosta translation files).